lib/Core/ExecutorUtil.cpp

```diff
@@ -58,9 +62,11 @@ namespace klee {
         return ConstantExpr::create(0, getWidthForLLVMType(c->getType()));
       } else if (const ConstantDataSequential *cds =
                  dyn_cast<ConstantDataSequential>(c)) {
+        // Handle a vector or array: first element has the smallest address,
```

Comment:
"address" is not very helpful to a reader here. The direction an address grows in depends on the platform.

Has reading constant arrays always been broken in KLEE then? If so we need more IR tests.

Reply:
@delcypher apparently yes. We do need more IR tests, but I would add them in a separate PR since they require more planning and work.

Reply:
@delcypher Why is "address" not helpful? The entry is endianness dependent but for subsequent elements, they have increasing addresses.

And yes, it looks like it has been broken but it was not utilized a lot I guess. With newer versions, additional optimizations utilize it more. So, yes, more IR tests would be great.